### PR TITLE
Merchant add item missing fields us #49 issue 23

### LINKED ISSUE #23: User Story 49, Merchant cannot add an item if details are bad/missing
```
As a merchant
When I try to add a new item
If any of my data is incorrect or missing (except image)
Then I am returned to the form
I see one or more flash messages indicating each error I caused
All fields are re-populated with my previous data
```

DO THESE STORIES FIRST:
- User Story 48, Merchant adds an item

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #29

child of #82

depends on #24

### PR DESCRIPTION
**Pull request by n-flint**
- [] Check this if the PR has been approved by the team to be a quick patch and the rest of this form will not be filled out

# Test Coverage Percentage (Model/Feature):
- rspec - 100.00%
- rspec spec/models - 98.29%

# What functionality does this accomplish?
Makes sure a merchant can only add an item if necessary fields are filled.

# What did you struggle on to complete?
Testing an image within a css tag, ended up just testing the actual image_url attribute of the new item.

# Implements/Fixes:

closes #23

# Testing Changes
- [x] No Tests have been changed
- [] Some Tests have been changed
- [] All of the Tests have been changed(Please describe what in the world happened)

# Checklist:

- [x] My code has no unused/commented out code
- [x] I have reviewed my code
- [] I have commented my code, particularly in hard-to-understand areas
- [x] I have fully tested my code

# Please include an emoji of how you feel about this branch:

### CODE REVIEW

File: spec/features/merchants/adds_items_spec.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+      fill_in "Quantity",	with: 100 
+      fill_in "Price",	with: 1.00
+
+      click_button "Add Item"
+
+      item = Item.last
+
+      expect(page).to have_content("Item Test is now saved and available for sale.")
+      
+      expect(item.image_url).to eq("https://i.pinimg.com/originals/2a/84/90/2a849069c7f487f71bb6594dffb84e5e.png")
```
Comment (n-flint) [inline spec/features/merchants/adds_items_spec.rb]: I was running into issues when I was originally testing this. I figured that if multiple items have a default image, then I would have to test within the css tag of the item I was testing. Could not get that to work, or even without the within block, and just testing the whole page for the image. I decided to just test the actual attribute of the item. It is working, I just think it could be better tested. Just letting you guys know!
